extensions-core/avro-extensions/src/main/java/io/druid/data/input/avro/SimpleAvroBytesDecoder.java

```diff
@@ -0,0 +1,86 @@
+/*
+ *
+ *  Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ *  or more contributor license agreements. See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership. Metamarkets licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License. You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ * /
+ *
+ */
+
+package io.druid.data.input.avro;
+
+import com.fasterxml.jackson.annotation.JacksonInject;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Preconditions;
+import com.metamx.common.logger.Logger;
+import com.metamx.common.parsers.ParseException;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.DatumReader;
+import org.apache.avro.io.DecoderFactory;
+import org.apache.avro.util.ByteBufferInputStream;
+
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ */
+public class SimpleAvroBytesDecoder implements AvroBytesDecoder
+{
+  private static final Logger logger = new Logger(SimpleAvroBytesDecoder.class);
+
+  private final Schema schemaObj;
+
+  private Map<String, Object> schema;
+
+  @JsonCreator
+  public SimpleAvroBytesDecoder(
+      @JacksonInject ObjectMapper mapper,
+      @JsonProperty("schema") Map<String, Object> schema
+  ) throws Exception
+  {
+    Preconditions.checkArgument(schema != null, "schema must be provided");
+
+    this.schema = schema;
+    String schemaStr = mapper.writeValueAsString(schema);;
+
+    logger.info("Schema string [%s]", schemaStr);
+    schemaObj = new Schema.Parser().parse(schemaStr);
```

Comment:
for that we can rely on avro lib only , if there is an error in the schema parsing, then this should throw exceptions.
otoh if the schema is ok but datum given in the parse is not deserializable then we should get an exception there.